### PR TITLE
Update the configs of ESLint to FlatConfig (and v9.x)

### LINKED ISSUE #391: Update the configs of ESLint to FlatConfig (and v9.x)
### Subject

Update the configs of ESLint to FlatConfig (and v9.x) 

### Description

Do the same thing you worked on with Yamada Theme

https://github.com/yamada-ui/yamada-theme/pull/16

### Are you willing to participate in the development of this feature and create a pull request?

None

### PR DESCRIPTION
**Pull request by illionillion**
Closes #391

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,17 +72,26 @@
     "chalk": "^5.3.0",
     "commander": "^12.1.0",
     "dotenv": "^16.4.5",
-    "eslint": "^8.57.0",
+    "eslint": "^9.8.0",
     "eslint-config-next": "^14.2.5",
```
Comment (koralle) [inline package.json:76]: @illionillion 

You can remove `eslint-config-next`, `eslint-config-prettier`.

Comment (illionillion) [reply package.json]: fixed

**Review by koralle — DISMISSED** [review]

**Comment by illionillion** [comment]
@koralle 
I resolved conflict.
Please check again.

**Comment by illionillion** [comment]
@koralle @108yen 
I resolved conflict and add rule.
Please review again.

File: package.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     "build": "next build",
     "build:profile": "next build --profile",
     "start": "next start",
-    "lint": "next lint",
+    "lint": "eslint . --max-warnings=0 --config eslint.config.mjs",
```
Comment (108yen) [inline package.json:15]: Please fix.
```json
    "lint": "eslint . --max-warnings=0 --cache",
```

**Review by 108yen — COMMENTED** [review]
Conflicts again.

**Comment by illionillion** [comment]
@108yen 
Fixed again.
Please review again.

**Review by 108yen — DISMISSED** [review]

**Review by koralle — DISMISSED** [review]
@illionillion 

Thank you!

For Yamada Components, it seems that simply writing `project: true` is sufficient for this part. 

```ts
project: `${dirname(fileURLToPath(import.meta.url))}/tsconfig.json`,
```

I just learned about this recently. For more details, please check the following article:

[Relative TSConfig Projects with `parserOptions.project = true` | typescript-eslint](https://typescript-eslint.io/blog/parser-options-project-true/)

**Comment by hirotomoyamada** [comment]
Close to create a new PR.

**Pull request closed without merging**